language-server/src/dotty/tools/languageserver/worksheet/ReplReader.scala

```diff
@@ -0,0 +1,47 @@
+package dotty.tools.languageserver.worksheet
+
+import java.io.{InputStream, InputStreamReader}
+
+/**
+ * Reads the output from the REPL and makes it available via `next()`.
+ *
+ * @param stream The stream of messages coming out of the REPL.
+ */
+private class ReplReader(stream: InputStream) extends Thread {
+  private val in = new InputStreamReader(stream)
+
+  private[this] var output: Option[String] = None
+  private[this] var closed: Boolean = false
+
+  override def run(): Unit = synchronized {
+    val prompt = "scala> "
+    val buffer = new StringBuilder
+    val chars = new Array[Char](256)
+    var read = 0
+
+    while (!Thread.interrupted() && { read = in.read(chars); read >= 0 }) {
+      buffer.appendAll(chars, 0, read)
+      if (buffer.endsWith(prompt)) {
```

Comment:
Isn't this true if I do println("scala> ") ? This is OK for now but eventually we should have a real REPL protocol (I have a student who's probably going to work on that).